ReSwift/CoreTypes/StoreType.swift

```diff
@@ -18,9 +18,6 @@ public protocol StoreType: DispatchingStoreType {
 
     associatedtype State: StateType
 
-    /// The current state stored in the store.
-    var state: State! { get }
```

Comment:
I don't think this change should be included in this PR. There are valid reasons for making `state` unreadable, but there are also valid reasons for reading state outside of a subscriber (IE one-time reads). Whether or not it stays I think this should be discussed & dealt with separately from the change to removing write access.

Reply:
I agree with that. Making the state totally private from the store could be a source of a lot of problem in some architectures.

Reply:
The original Redux implementation in JavaScript has a `getState()` method to query the state of the store at any time, so that should be enough, IMO!

Reply:
Instead of Java-style getters like `getState()`, making the property puplicly readable via `public private(set)` could work.

Reply:
Would be perfect. Setter can and should be private anyway.

Reply:
Yeah, they probably do it because of JavaScript, but in Swift we could just use the read-only variable. It's the same as in the middleware, you get both the `dispatch` function and `getState`.